backend/modules/model_gateway/model_gateway.py

```diff
@@ -1,85 +1,119 @@
 import json
-from pydantic import BaseModel
-from typing import List, Optional
-from backend.types import ModelConfig, ModelProviderConfig, ModelType
+import os
+from typing import List
+
 from langchain.embeddings.base import Embeddings
-from langchain_openai import OpenAIEmbeddings
 from langchain_core.language_models.chat_models import BaseChatModel
+from langchain_openai import OpenAIEmbeddings
 from langchain_openai.chat_models import ChatOpenAI
-import os
+
+from backend.logger import logger
+from backend.types import ModelConfig, ModelProviderConfig, ModelType
+
 
 class ModelGateway:
     config: List[ModelProviderConfig]
     modelsToProviderMap = {}
-    def __init__(self):        
+
+    def __init__(self):
         with open("./models_config.json") as f:
             data = json.load(f)
             self.config = [ModelProviderConfig(**item) for item in data]
             self.models: List[ModelConfig] = []
             for list in self.config:
-                if list.api_key_env_var and not os.environ.get(list.api_key_env_var):
-                    raise ValueError(f"Environment variable {list.api_key_env_var} not set.")
                 for model_id in list.embedding_model_ids:
-                    model_name = f"{list.provider_name}/{model_id}"
-                    self.modelsToProviderMap[model_name] = list
+                    if list.api_key_env_var and os.environ.get(list.api_key_env_var):
+                        # Register only those models whose API key is set
+                        model_name = f"{list.provider_name}/{model_id}"
+                        logger.info(f"Adding model {model_name} to the model gateway.")
+                        self.modelsToProviderMap[model_name] = list
+                    else:
+                        # Avoid adding the model to the map if the API key is not set
+                        logger.warning(
+                            f"Environment variable {list.api_key_env_var} not set."
```

Comment:
This error message should be more helpful
We should inform we are skipping adding the model to list of models

Reply:
done.